### PR TITLE
avoid calling OnFlowDelivery with nil

### PR DESCRIPTION
**Pull request by kaworu**
See commits.

### CODE REVIEW

File: pkg/hubble/observer/local_observer.go [hunk 1]
```diff
@@ -283,6 +283,15 @@ nextEvent:
 		switch ev := e.Event.(type) {
 		case *flowpb.Flow:
 			eventsReader.eventCount++
```
Comment (kaworu) [inline pkg/hubble/observer/local_observer.go:285]: @tklauser do you know why `eventCount` is only incremented when the event is a flow (i.e. it's not incremented for a `LostEvent` for example)?

**Review by pchaigno — APPROVED** [review]
Does the bugfix in the last commit need to be backported? Is there any user-visible impact?

**Review by rolinh — APPROVED** [review]

File: pkg/hubble/observer/local_observer.go (hunk 1 above)
Comment (rolinh) [reply pkg/hubble/observer/local_observer.go:285]: `LostEvent` is a special case and if you need to ask then it probably deserves a comment there in the switch case. A `LostEvent` is an event type that is never explicitly requested by the user (say, a query asks for 20 events, then lost events should not be accounted for as they are not events per se but an indication that some event was lost). The Hubble CLI actually prints lost events to stderr iirc.
You can also have a look at the comments and LostEvent being treated as a special case in `newRingReader()` and `eventsReader.Next()` further below.

**Comment by kaworu** [comment]
@pchaigno 
> Does the bugfix in the last commit need to be backported?

Good question, @michi-covalent what do you think?

>  Is there any user-visible impact?

None outside `OnFlowDelivery` consumers.

File: pkg/hubble/observer/local_observer.go (hunk 1 above)
Comment (tklauser) [reply pkg/hubble/observer/local_observer.go:285]: Thanks for clarifying @rolinh!

**Review by tklauser — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]

File: pkg/hubble/observer/local_observer.go (hunk 1 above)
Comment (kaworu) [reply pkg/hubble/observer/local_observer.go:285]: Thanks for the explanation @rolinh, I've added a comment in the LostEvent case.

**Pull request merged**